index.js

```diff
@@ -32,21 +43,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var child_process = spawn(nativeBinary, args, {stdio: stdio});
-  child_process.on('close', function(exit) {
+  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
+  clangFormatProcess.on('close', function(exit) {
     if (exit) done(exit);
   });
-  return child_process;
+  return clangFormatProcess;
 }
 
-module.exports = clangFormat;
-module.exports.spawnClangFormat = spawnClangFormat;
-
-if (require.main === module) {
+function main() {
+  // Find clang-format in node_modules of the project of the .js file, or cwd.
+  var nonDashArgs = process.argv.filter(function(arg, idx) { return idx > 1 && arg[0] != '-' });
```

Comment:
I manually tested. Let me take another stab at an automated test (maybe in a subsequent commit), the first attempt ended up being pretty nasty.